src/auth/basic/Config.cc

```diff
@@ -109,7 +109,6 @@ Auth::Basic::Config::done()
         helperShutdown(basicauthenticators);
     }
 
-    delete basicauthenticators;
     basicauthenticators = nullptr;
```

Comment:
> PR description: It is possible that Squid already tried to work around these problems by not destructing some helper objects during the shutdown phase of their reconfiguration:

* Auth::{Digest,Negotiate,Ntlm}::Config::done(), redirectShutdown(), and externalAclShutdown() keep stale helper objects alive, resulting in excessive helperShutdown() calls (in some cases) and possibly worse problems.
* However, that ugly workaround or accident does not apply to Auth::Basic::Config::done(), Ssl::Helper::Shutdown(), and Ssl::CertValidationHelper::Shutdown() code, resulting (until this fix) in accesses to destructed "parent" objects.

As discussed, we need to figure out why these two groups of functions still differ (at least on the surface) in how they treat the corresponding "global" pointers to those allegedly "stale" helper objects. PR description and/or code changes are needed to remove and/or document/clarify this difference. PR description changes are needed to clarify whether those "excessive helperShutdown() calls (in some cases) and possibly worse problems" still exist in PR code.